pims/image_sequence.py

```diff
@@ -332,3 +332,33 @@ def __repr__(self):
                                   dtype=self.pixel_type,
                                   C=self._sizeC,
                                   Z=self._sizeZ)
+
+
+def customize_image_sequence(imread_func, name=None):
+    """Class factory for ImageSequence with customized image reader.
+
```

Comment:
Neat. So users can write their own reader function, like image reader or hdf5 reader, and use it here. Similar to the idea of file store, or file store borrows some idea here?